db/transaction_log_impl.h

```diff
@@ -54,14 +55,68 @@ class LogFileImpl : public LogFile {
   uint64_t sizeFileBytes_;
 };
 
+class TransactionLogSeqCache {
+ public:
+  TransactionLogSeqCache(size_t size)
+      : size_(size), rand_(std::random_device{}()) {}
+  struct SeqWithFileBlockIdx {
+    uint64_t log_number;
+    uint64_t seq_number;
+    uint64_t block_index;
+    SeqWithFileBlockIdx(uint64_t log_num, uint64_t seq_num, uint64_t block_idx)
+        : log_number(log_num), seq_number(seq_num), block_index(block_idx){};
+    bool operator<(const SeqWithFileBlockIdx& other) const {
+      return std::tie(other.log_number, other.seq_number, other.block_index) <
+             std::tie(log_number, seq_number, block_index);
+    }
+    bool operator==(const SeqWithFileBlockIdx& other) const {
+      return std::tie(log_number, seq_number, block_index) ==
+             std::tie(other.log_number, other.seq_number, other.block_index);
+    }
+  };
+
+  void Insert(uint64_t log_number, uint64_t seq_number, uint64_t block_index) {
+    std::lock_guard<std::mutex> lk{mutex_};
+    if (cache_.size() > size_) {
+      auto iter = cache_.begin();
+      std::advance(iter, rand_.Next() % cache_.size());
```

Comment:
What about always erasing the oldest one? I assume GetUpdatesSince() is called with likely increasing sequence numbers.

Reply:
has changed to replace the oldest one. thx for the advice